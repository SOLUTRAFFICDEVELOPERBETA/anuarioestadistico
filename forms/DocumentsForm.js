import React from 'react';
import PropTypes from 'prop-types';
import {
    AppBar,
    Toolbar,
    Typography,
    Tooltip,
    IconButton,
    Popover,
    List,
    ListItem,
    ListItemIcon,
    ListItemText,
    makeStyles,
    Dialog,
    DialogTitle,
    DialogContent
} from '@material-ui/core';
import { MoreVert, PostAddTwoTone } from '@material-ui/icons';
import SearchInput from '../components/SearchInput';
import { documents } from '../constants/templates';

// Estilos del componente
const useStyles = makeStyles((theme) => ({
    appbar: {
        marginY: theme.spacing(5)
    },
    toolbar: {
        display: 'flex'
    }
}));
/**
 * @description Componente que permite al usuario selección el tipo de formato en el momento de crear la páginal.
 * @param {func: () => void} onSubmit Propiedades del componente.
 * Consulte los elementos utilizados en {@link https://material-ui.com/ | Material-ui}
 */
const DocumentsForm = ({ onSubmit }) => {
    const classes = useStyles();
    const [openForm, setOpenForm] = React.useState(false);
    const [optionsAnchorEl, setOptionsAnchorEl] = React.useState(null);

    const handleOpenOptions = (event) => {
        setOptionsAnchorEl(event.currentTarget);
    };

    const handleCloseOptions = () => {
        setOptionsAnchorEl(null);
    };

    const handleSubmit = (temp) => {
        setOpenForm(false);
        onSubmit(temp);
    };

    const handleSearch = ({ search }) => {
        const array = [...templates].filter((temp) =>
            temp.title.toLowerCase().includes(search.toLowerCase())
        );
        setTemplateList(array);
    };

    const openOptions = Boolean(optionsAnchorEl);


    return (
        <React.Fragment>
            <AppBar position="static" color="transparent" elevation={0} className={classes.appbar}>
                <Toolbar variant="dense" className={classes.toolbar}>
                    <Typography color="primary" align="left" variant="h4" style={{ flexGrow: 1 }}>
                        Administración de Páginas
                    </Typography>
                    <Tooltip title="Opciones">
                        <IconButton size="medium" color="primary" onClick={handleOpenOptions}>
                            <MoreVert />
                        </IconButton>
                    </Tooltip>
                    {openOptions && (
                        <Popover
                            open={openOptions}
                            anchorEl={optionsAnchorEl}
                            onClose={handleCloseOptions}
                            anchorOrigin={{
                                vertical: 'bottom',
                                horizontal: 'center'
                            }}
                            transformOrigin={{
                                vertical: 'top',
                                horizontal: 'right'
                            }}>
                            <List dense>
                                <ListItem button onClick={() => setOpenForm(true)}>
                                    <ListItemIcon>
                                        <PostAddTwoTone />
                                    </ListItemIcon>
                                    <ListItemText primary="Añadir Página" />
                                </ListItem>
                            </List>
                        </Popover>
                    )}
                </Toolbar>
            </AppBar>
            {openForm && (
                <Dialog
                    open={openForm}
                    onClose={() => setOpenForm(false)}
                    scroll="paper"
                    fullWidth
                    maxWidth="lg">
                    <DialogTitle>selecciona la Página a crear</DialogTitle>
                    <DialogContent>
                        <SearchInput onSearch={handleSearch} />
                        <List>
                            {documents.map((item, key) => (
                                <ListItem key={key} button divider onClick={() => handleSubmit(item)}>
                                    <ListItemText primary={item.title} />
                                </ListItem>
                            ))}
                        </List>
                    </DialogContent>
                </Dialog>
            )}
        </React.Fragment>
    );
};

DocumentsForm.propTypes = {
    onSubmit: PropTypes.func
};

export default DocumentsForm;
