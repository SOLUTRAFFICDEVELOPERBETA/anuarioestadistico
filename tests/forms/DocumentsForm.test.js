import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import DocumentsForm from '../../forms/DocumentsForm';
import { act } from '@testing-library/react';


describe('Pruebas en <DocumentsForm/>', () => {
    const onSubmit = jest.fn();


    let wrapper;
    beforeEach(() => {
        wrapper = shallow(
            <DocumentsForm
                onSubmit={onSubmit}
            />
        )
        jest.clearAllMocks()
    })
    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe llamar la funciÃ³n onSubmit()', () => {
        act(() => {
            wrapper.find('WithStyles(ForwardRef(IconButton))').prop('onClick')({
                currentTarget: {

                    innerText: 'innerText prueba de texto',
                    localName: "span"

                }
            });
        })
        wrapper.find('WithStyles(ForwardRef(ListItem))').prop('onClick')({
            currentTarget: {
                innerText: 'innerText prueba de texto TEMPRORAL',
                localName: "span"
            }
        })
        wrapper.find('WithStyles(ForwardRef(ListItem))').at(1).prop('onClick')();
        expect(onSubmit).toHaveBeenCalled();
        expect(onSubmit).toHaveBeenCalledTimes(1);
    })
})