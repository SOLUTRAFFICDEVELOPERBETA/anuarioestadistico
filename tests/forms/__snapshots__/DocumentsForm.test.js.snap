// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DocumentsForm/> debe mostrarse correctamente 1`] = `
<Fragment>
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-appbar-1"
    color="transparent"
    elevation={0}
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))
      className="makeStyles-toolbar-2"
      variant="dense"
    >
      <WithStyles(ForwardRef(Typography))
        align="left"
        color="primary"
        style={
          Object {
            "flexGrow": 1,
          }
        }
        variant="h4"
      >
        Administración de Páginas
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Tooltip))
        title="Opciones"
      >
        <WithStyles(ForwardRef(IconButton))
          color="primary"
          onClick={[Function]}
          size="medium"
        >
          <MoreVertIcon />
        </WithStyles(ForwardRef(IconButton))>
      </WithStyles(ForwardRef(Tooltip))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</Fragment>
`;
